Tentacle/Client.swift

```diff
@@ -55,6 +54,18 @@ extension URLRequest {
         
         return request
     }
+
+    internal static func create<Request: RequestType>(_ url: URL, _ request: Request, _ credentials: Client.Credentials?, contentType: String? = Client.APIContentType) -> URLRequest {
```

Comment:
The problem I have with an optional Request is the compiler not being able to infer the type. Any idea how to avoid that cleanly?